examples/experimental/dagster-airlift/dagster_airlift/in_airflow/materialize_assets_operator.py

```diff
@@ -0,0 +1,47 @@
+import os
+from typing import Any, Iterable, Mapping, Sequence, Union
+
+import requests
+from airflow.utils.context import Context
+
+from dagster_airlift.in_airflow.base_asset_operator import BaseDagsterAssetsOperator
+
+
+class BaseMaterializeAssetsOperator(BaseDagsterAssetsOperator):
+    """An operator base class that proxies execution to a user-provided list of Dagster assets.
+    Will throw an error at runtime if not all assets can be found on the corresponding Dagster instance.
+    """
+
+    def __init__(self, asset_key_paths: Sequence[Union[str, Sequence[str]]], *args, **kwargs):
+        self.asset_key_paths = [
+            [path] if isinstance(path, str) else path for path in asset_key_paths
+        ]
+        super().__init__(*args, **kwargs)
+
+    def filter_asset_nodes(
+        self, context: Context, asset_nodes: Sequence[Mapping[str, Any]]
+    ) -> Iterable[Mapping[str, Any]]:
+        path_str_to_node = {
+            get_asset_key_str(node["assetKey"]["path"]): node for node in asset_nodes
+        }
+        if not all(get_asset_key_str(path) in path_str_to_node for path in self.asset_key_paths):
+            raise ValueError(
+                f"Could not find all asset key paths {self.asset_key_paths} in the asset nodes. Found: {list(path_str_to_node.keys())}"
+            )
+        yield from [path_str_to_node[get_asset_key_str(path)] for path in self.asset_key_paths]
+
+
+def get_asset_key_str(path: Sequence[str]) -> str:
+    return "/".join(path)
```

Comment:
This isn't safe

AssetKey("//", "/") and AssetKey("/", "//") would cause a collision